types/container/host_config.go

```diff
@@ -141,25 +141,44 @@ type PidMode string
 
 // IsPrivate indicates whether the container uses its private pid stack.
 func (n PidMode) IsPrivate() bool {
-	return !(n.IsHost())
+	return !(n.IsHost() || n.IsContainer())
 }
 
 // IsHost indicates whether the container uses the host's pid stack.
 func (n PidMode) IsHost() bool {
 	return n == "host"
 }
 
+// IsContainer indicates whether the container uses a container's pid stack.
```

Comment:
I'm pretty sure that stack is from "network stack" :) I don't think that it's applicable here.